video/out/cocoa_common.m

```diff
@@ -574,7 +574,6 @@ int vo_cocoa_config_window(struct vo *vo)
         if (!s->embedded && s->window) {
             if (reset_size)
                 queue_new_video_size(vo, width, height);
-            vo_cocoa_fullscreen(vo);
```

Comment:
the actual problem imo is not the `vo_cocoa_fullscreen` call but some calls inside that function. specifically `vo_cocoa_update_screen_info()` and `flag_events(vo,VO_EVENT_ICC_PROFILE_CHANGED)`. former is already called at the beginning of `vo_cocoa_config_window` and just leads to redundancy, latter shouldn't be handled by the fullscreen code i would say (but i removed that in one of the commits). 

it's probably better to remove that commit and change `vo_cocoa_fullscreen` a bit?